### DISCUSSION TITLE
react-native autolink dependencies used in custom library

### DISCUSSION THREAD

**Discussion by Rajesh6381** [community/community#141238] [Programming Help]
### Select Topic Area

General

### Body

So I build a library, and I am using some external libraries in it.

This is the package.json

` "peerDependencies": {
    "@react-native-async-storage/async-storage": "*",
    "crypto-js": "*",
    "js-base64": "*",
    "react-native": "^0.0.0-0 || 0.60 - 0.69 || 1000.0.0",
    "react-native-sqlite-2": "*",
    "react-native-windows": "*"
  },`

Now when I install my library, the peerDependencies gets installed automatically.

The problem is that those installed automatically dose not autolink. for example for react-native-sqlite-2 to link, I have to installed manually to eg npm install react-native-sqlite-2

Why is that? should those libraries not autolink by them self? or is there a configuration that I am missing ?

**Comment by ovftank** [comment]
Hi @Rajesh6381,

The issue you're encountering is due to the fact that **auto-linking in React Native only works with dependencies installed directly in the project**, not with **peerDependencies**.

### Explanation:
- **peerDependencies** are meant to indicate the versions of libraries that your library is compatible with, but they are **not installed automatically** when your library is installed.
- React Native's **auto-linking** system works with packages that are listed under **dependencies** in the project’s `package.json`. If your libraries like `react-native-sqlite-2` are listed as `peerDependencies`, they won’t be installed automatically, and React Native won't auto-link them.

### Solution:
To fix the issue, you need to **Move the libraries to `dependencies`** in your library's `package.json` so that they are installed and auto-linked when your library is installed!

**Comment by Rajesh6381** [comment]
Hi @ovftank,

I have changed my peer dependencies to dependencies, but the autolinking is still not working. I also tried another approach to autolink the library:

javascript
Copy code
 'react-native-sqlite-2': {
     root: path.join(__dirname, 'node_modules/react-native-sqlite-2'),
 }
I added the library path in the react-native.config.js under the dependencies section in project. This works fine, and the autolinking is functioning for iOS, but it is not working for Android.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @Rajesh6381 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
